src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -481,38 +477,100 @@ Namespace Microsoft.CodeAnalysis.Operations
+                       nestedConversion.ExplicitCastInCode AndAlso
+                       nestedConversion.Type <> nestedOperand.Type AndAlso
+                       nestedConversion.Type = boundConversion.Type Then
+                        conversion = CreateConversion(nestedConversion)
+                        operandLazy = CreateNestedConversionOperand(operand, boundConversion.Syntax)
+                        operand = nestedOperand
+                    End If
+                ElseIf nestedOperand.Syntax.Kind() = SyntaxKind.AddressOfExpression AndAlso
+                       boundConversion.Type = nestedOperand.Type AndAlso
+                       IsDelegateCreation(boundConversion.Syntax, nestedOperand, nestedOperand.Type) Then
+                    conversion = New Conversion(KeyValuePair.Create(Of ConversionKind, MethodSymbol)(boundConversion.ConversionKind, Nothing))
+                    operandLazy = CreateNestedConversionOperand(operand, boundConversion.Syntax)
+                    operand = nestedOperand
                 End If
-                Return (operandLazy, IsDelegateCreation:=True)
-            Else
-                Dim methodSymbol As MethodSymbol = Nothing
-                Return (New Lazy(Of IOperation)(Function() Create(operand)), IsDelegateCreation:=False)
             End If
+
+            If operandLazy Is Nothing Then
+                operandLazy = New Lazy(Of IOperation)(Function() CreateConversionOperand(operand))
+            End If
+
+            Return (operandLazy, conversion, IsDelegateCreation(boundConversion.Syntax, operand, boundConversion.Type))
         End Function
 
         ''' <summary>
         ''' Gets the operand and method symbol from a BoundConversion, compensating for if the conversion is a user-defined conversion
         ''' </summary>
-        Private Shared Function GetConversionInfo(boundConversion As BoundConversion) As (Operand As BoundExpression, Method As MethodSymbol)
+        Private Shared Function GetConversionOperand(boundConversion As BoundConversionOrCast) As BoundExpression
             If (boundConversion.ConversionKind And ConversionKind.UserDefined) = ConversionKind.UserDefined Then
                 Dim userDefinedConversion = DirectCast(boundConversion.Operand, BoundUserDefinedConversion)
-                Return (Operand:=userDefinedConversion.Operand, Method:=userDefinedConversion.Call.Method)
+                Return userDefinedConversion.Operand
+            Else
+                Return boundConversion.Operand
+            End If
+        End Function
+
+        Private Function CreateNestedConversionOperand(operand As BoundExpression, conversionSyntax As SyntaxNode) As Lazy(Of IOperation)
+            Return New Lazy(Of IOperation)(Function() CreateNestedConversionOperandCore(operand))
+        End Function
+
+        Private Function CreateNestedConversionOperandCore(operand As BoundExpression) As IOperation
```

Comment:
>CreateNestedConversionOperandCore [](start = 25, length = 33)

It feels like we are duplicating some of the logic that we already did in the caller. This means that we need to keep both functions in sync, which feel fragile and bug prone.